docs/book/start/logging-packets.md

```diff
@@ -87,73 +87,50 @@ The data structure that we'll need to send information to user-space will need t
 
 ## Writing Data
 
-### Generating Bindings To vmlinux.h
+### Using Kernel Network Types
 
-To get useful data to add to our maps, we first need some useful data structures to populate with data from the `XdpContext`.
+To get useful data to add to our maps, we first need some useful data structures
+to populate with data from the `XdpContext`.
 We want to log the Source IP Address of incoming traffic, so we'll need to:
 
 1. Read the Ethernet Header to determine if this is an IPv4 Packet
 1. Read the Source IP Address from the IPv4 Header
 
-The two structs in the kernel for this are `ethhdr` from `uapi/linux/if_ether.h` and `iphdr` from `uapi/linux/ip.h`.
-If I were to use bindgen to generate Rust bindings for those headers, I'd be tied to the kernel version of the system that I'm developing on.
-This is where `aya-tool` comes in to play. It can easily generate bindings for using the BTF information in `/sys/kernel/btf/vmlinux`.
+The two structs in the kernel for this are `ethhdr` from `uapi/linux/if_ether.h`
+and `iphdr` from `uapi/linux/ip.h`. Rust equivalents of those structures (`EthHdr`
+and `Ipv4Hdr`) are provided by the [network-types crate](https://crates.io/crates/network-types).
 
-First, we must make sure that `bindgen` is installed.
-```sh
-cargo install bindgen-cli
-```
-
-Once the bindings are generated and checked in to our repository they shouldn't need to be regenerated again unless we need to add a new struct.
-
-Lets use `xtask` to automate this so we can easily reproduce this file in future.
-
-We'll add the following code
-
-=== "xtask/src/codegen.rs"
-
-    ```rust linenums="1"
-    --8<-- "examples/myapp-02/xtask/src/codegen.rs"
-    ```
+Let's add it to our eBPF crate by adding a dependency on `network-types = "0.0.2"`
```

Comment:
These kinds of versions can get lost in time and cause issues for readers. Has there been any previous thoughts on doing something like templating from the `Cargo.toml` data in the relevant examples? Not at all something to hold up this PR for of course, but if there's any interest maybe I can throw together an issue for it.

Reply:
Good point. I removed the version from the sentence, we show the whole `Cargo.toml` content below anyway, so I think that should suffice.

Reply:
done